ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Basics.hs

```diff
@@ -197,6 +200,19 @@ class ( -- Requirements on the ledger state itself
     -> LedgerResult l (Ticked1 l TrackingMK)
 
 
+  -- | Given a block, get the key-sets that we need to apply it to a ledger
+  -- state.
+  --
+  -- TODO: the assumption here is that the the resulting key-sets contain the
+  -- keys that the ledger needs to tick to the slot of the given block.
+  --
+  -- TODO: this might not be the best place to define this function. Maybe we
+  -- want to make the on-disk ledger state storage concern orthogonal to the
+  -- ledger state transformation concern.
+  getKeySets :: blk -> Maybe (l EmptyMK) -> TableKeySets l
```

Comment:
Maybe remove the `Maybe (l ...)` argument. Another option is to remove the `Maybe`.

We could implement `getBlockKeySets` in terms of these two functions:
```haskell
getTxsKeySets :: [txs] -> TableKeySets l

--  We won't use this in the first iteration.
getTickKeySets :: SlotNo -> l EmptyMk  -> TableKeySets l
```

Reply:
We'd need `getTxsKeySets` in the mempool.